cpp/tests/binaryop/util/operation.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Copyright 2018-2019 BlazingDB, Inc.
+ *     Copyright 2018 Christian Noboa Mardini <christian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cmath>
+#include <type_traits>
+#include <cstdint>
+
+namespace cudf {
+namespace library {
+namespace operation {
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Add {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs + (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Sub {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs - (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Mul {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs * (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct Div {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            using TypeCommon = typename std::common_type<TypeLhs, TypeRhs>::type;
+            return (TypeOut)((TypeCommon)lhs / (TypeCommon)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct TrueDiv {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
+            return (TypeOut)((double)lhs / (double)rhs);
+        }
+    };
+
+    template <typename TypeOut, typename TypeLhs, typename TypeRhs>
+    struct FloorDiv {
+        TypeOut operator()(TypeLhs lhs, TypeRhs rhs) {
```

Comment:
This is a method of a stateless object. No reason why it shouldn't be `const`. 

Also, please consider using explicit C++-style casts `static_cast<>, const_cast<>, dynamic_cast<>`, etc.